src/kakarot/library.cairo

```diff
@@ -191,9 +192,8 @@ namespace Kakarot {
     ) {
         alloc_locals;
         let (current_salt) = salt.read();
-        let (evm_contract_address, starknet_contract_address) = ContractAccount.deploy(
-            current_salt
-        );
+        let (evm_contract_address) = CreateHelper.get_create_address(0, current_salt);
```

Comment:
use `IAccount.get_eth_address(caller)` instead of `0`, (or add a TODO and an issue)

Reply:
Nice. This was a bit I was unsure on!